src/pages/challenges.tsx

```diff
@@ -30,25 +36,27 @@ const Challenges: FC = () => {
   }, []);
 
   useEffect(() => {
-    if ('challenges' in response) {
-      let challenges: challenge['content'] = response.challenges;
-      const challengeMap: any = [];
-
-      challengeMap.Active = challenges.filter((task) => task.is_active);
-      challengeMap.Completed = challenges.filter((task) => !task.is_active);
-
-      setFilteredChallenge(challengeMap);
+    if (isLoggedIn && !Object.keys(response).length) {
+      callAPI();
+      if ('challenges' in response) {
+        let challenges: challenge['content'] = response.challenges;
+        const challengeMap: any = [];
+  
+        challengeMap.Active = challenges.filter((task) => task.is_active);
+        challengeMap.Completed = challenges.filter((task) => !task.is_active);
+  
+        setFilteredChallenge(challengeMap);
+      }
     }
-  }, [isLoading, response]);
+  }, [isLoggedIn, response])
```

Comment:
Why has is loading been removed

Reply:
We are checking if the user is logged in first. Only if they are logged in can the loading be done now